### PR TITLE
[Release-0.15] Switch from the NoFork terminology to the TerraformPluginSDK terminology and bump dependencies

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

In this PR:
- Switch from the NoFork terminology to the TerraformPluginSDK terminology
- Bump crossplane-runtime version to `v1.15.1`
- Bump upjet version to `v1.1.1`

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

Tested with uptest below

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/applications/application.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/conditionalaccess/accesspolicy.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/groups/member.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/serviceprincipals/principal.yaml"

File: go.mod [hunk 1]
```diff
@@ -90,50 +92,60 @@ require (
...
+
+replace golang.org/x/exp => golang.org/x/exp v0.0.0-20230713183714-613f0c0eb8a1
+
+replace github.com/hashicorp/terraform-json => github.com/hashicorp/terraform-json v0.17.0
+
+replace github.com/hashicorp/terraform-plugin-sdk/v2 => github.com/hashicorp/terraform-plugin-sdk/v2 v2.26.1
+
+replace github.com/hashicorp/terraform-plugin-go => github.com/hashicorp/terraform-plugin-go v0.16.0
+
+replace github.com/hashicorp/terraform-plugin-framework => github.com/hashicorp/terraform-plugin-framework v1.2.0
```
Comment (ulucinar) [inline go.mod:151]: Let's fix the versions to the ones already defined in `go.mod`:
```suggestion

replace golang.org/x/exp => golang.org/x/exp v0.0.0-20231006140011-7918f672742d

replace github.com/hashicorp/terraform-json => github.com/hashicorp/terraform-json v0.16.0

replace github.com/hashicorp/terraform-plugin-sdk/v2 => github.com/hashicorp/terraform-plugin-sdk/v2 v2.26.1

replace github.com/hashicorp/terraform-plugin-go => github.com/hashicorp/terraform-plugin-go v0.14.3
```

**Review by ulucinar — APPROVED** [review]
Thanks @turkenf, lgtm.
Let's also run at least one uptest again after these dependency changes.

**Comment by turkenf** [comment]
/test-examples="examples/serviceprincipals/principal.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/groups/member.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/applications/application.yaml"

**Pull request merged**